fluidreview/signals.py

```diff
@@ -0,0 +1,33 @@
+"""Signals for fluidreview models"""
+import logging
+
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from rest_framework import serializers
+
+from fluidreview.api import FluidReviewAPI, process_user, parse_webhook
+from fluidreview.constants import WebhookParseStatus
+from fluidreview.models import WebhookRequest
+from fluidreview.serializers import UserSerializer
+from profiles.models import Profile
+
+log = logging.getLogger(__name__)
+
+
+@receiver(post_save, sender=WebhookRequest, dispatch_uid="webhookrequest_post_save")
+def handle_parse_webhook(sender, instance, created, **kwargs):  # pylint:disable=unused-argument
+    """
+    Extract individual attribute values from a WebhookRequest body; create/update User & Profile if necessary
+    """
+    if created and instance.status == WebhookParseStatus.CREATED:
+        parse_webhook(instance)
+        if instance.status == WebhookParseStatus.SUCCEEDED:
+            if not Profile.objects.filter(fluidreview_id=instance.user_id):
+                try:
+                    # Get user info from FluidReview API (ensures that webhook user_id is real).
+                    user_info = FluidReviewAPI().get('/users/{}'.format(instance.user_id)).json()
+                    serializer = UserSerializer(data=user_info)
+                    serializer.is_valid(raise_exception=True)
+                    process_user(serializer.data)
+                except (ConnectionError, serializers.ValidationError) as exc:
+                    log.error('Problem processing a user from WebhookRequest %s: %s', instance.id, str(exc))
```

Comment:
Can you use `log.exception` instead of `str(exc)` here?

Reply:
OK